ext/intl/bidi/bidi.c

```diff
@@ -284,43 +331,60 @@ static PHP_METHOD(IntlBidi, setContext) {
 /* }}} */
 #endif /* ICU >= 4.8 */
 
-/* {{{ proto self IntlBidi::setPara(string $paragraph[, int $paraLevel = IntlBidi::DEFAULT_LTR]) */
+/* {{{ proto self IntlBidi::setPara(string $paragraph[, int $paraLevel = IntlBidi::DEFAULT_LTR[, string $embeddingLevels]]) */
 ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX(bidi_setpara_arginfo, ZEND_RETURN_VALUE, 1, IS_OBJECT, 0)
 	ZEND_ARG_TYPE_INFO(0, paragraph, IS_STRING, 0)
 	ZEND_ARG_TYPE_INFO(0, paraLevel, IS_LONG, 0)
+	ZEND_ARG_TYPE_INFO(0, embeddingLevels, IS_STRING, 1)
 ZEND_END_ARG_INFO();
 static PHP_METHOD(IntlBidi, setPara) {
+	zend_string *para, *embeddingLevels = NULL;
+	zend_long paraLevel = UBIDI_DEFAULT_LTR;
 	php_intl_bidi_object *objval = bidi_object_from_zend_object(Z_OBJ_P(getThis()));
-	zend_string *para;
 	UChar *upara = NULL;
 	int32_t upara_len = 0;
-	zend_long paraLevel = UBIDI_DEFAULT_LTR;
 	UErrorCode error;
 
-	if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "S|l", &para, &paraLevel) == FAILURE) {
-		return;
+	if (objval->bidi->childCount > 1) {
+		bidi_free_bidi_object(objval->bidi);
+		objval->bidi = bidi_create_bidi_object(0, 0);
 	}
 
+	ZEND_PARSE_PARAMETERS_START(1, 3)
+		Z_PARAM_STR(para)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_LONG(paraLevel)
+		Z_PARAM_STR_EX(embeddingLevels, 0, 0)
```

Comment:
You've marked this string arg as "allows null" in the arg info struct, so you should parse it to match in this statement:

```suggestion
		Z_PARAM_STR_EX(embeddingLevels, 1, 0)
```

Reply:
Fixed it.